crates/trie/sparse/src/state.rs

```diff
@@ -124,11 +124,38 @@ impl SparseStateTrie {
         Ok(())
     }
 
+    /// Remove the leaf node.
+    pub fn remove_leaf(&mut self, path: &Nibbles) -> SparseStateTrieResult<()> {
+        self.state.remove_leaf(path)?;
+        Ok(())
+    }
+
     /// Returns sparse trie root if the trie has been revealed.
     pub fn root(&mut self) -> Option<B256> {
         self.state.root()
     }
 
+    /// Calculates the hashes of the nodes below the provided level.
+    pub fn calculate_below_level(&mut self, level: usize) {
+        self.state.calculate_below_level(level);
+    }
+
+    /// Update the leaf node of a storage trie at the provided address.
+    pub fn update_storage_leaf(
+        &mut self,
+        address: B256,
+        slot: Nibbles,
+        value: Vec<u8>,
+    ) -> SparseStateTrieResult<()> {
+        self.storages.entry(address).or_default().update_leaf(slot, value)?;
+        Ok(())
+    }
+
+    /// Wipe the storage trie at the provided address.
+    pub fn wipe_storage(&mut self, address: B256) {
+        self.storages.remove(&address);
```

Comment:
is this correct? we still need to preserve the fact that the storage was wiped and wipe it from the database on commit

Reply:
nvm it's still incorrect, turning into a draft for now